osgar/drivers/realsense.py

```diff
@@ -36,48 +39,101 @@ def t265_to_osgar_orientation(t265_orientation):
 class RealSense(Node):
     def __init__(self, config, bus):
         super().__init__(config, bus)
-        bus.register('pose2d', 'pose3d', 'raw', 'orientation')
+        bus.register('pose2d', 'pose3d', 'pose_raw', 'orientation', 'depth')
         self.verbose = config.get('verbose', False)
-        self.pipeline = None  # not initialized yet
-        self.pipeline_factory = rs.pipeline
-
-    def update(self):
-        channel = super().update()  # define self.time
-        if channel == 'trigger':
-            frames = self.pipeline.wait_for_frames()
-            pose_frame = frames.get_pose_frame()
-            if pose_frame:
-                pose = pose_frame.get_pose_data()
-                n = pose_frame.get_frame_number()
-                timestamp = pose_frame.get_timestamp()
-                orientation = t265_to_osgar_orientation(pose.rotation)
-                self.publish('orientation', orientation)
-                x, y, z = t265_to_osgar_position(pose.translation)
-                yaw = quaternion.heading(orientation)
-                self.publish('pose2d', [int(x*1000), int(y*1000), int(math.degrees(yaw)*100)])
-                self.publish('pose3d', [[x, y, z], orientation])
-                self.publish('raw', [n, timestamp,
-                    [pose.translation.x, pose.translation.y, pose.translation.z],
-                    [pose.rotation.x, pose.rotation.y, pose.rotation.z, pose.rotation.w],
-                    [pose.velocity.x, pose.velocity.y, pose.velocity.z],
-                    [pose.angular_velocity.x, pose.angular_velocity.y, pose.angular_velocity.z],
-                    [pose.acceleration.x, pose.acceleration.y, pose.acceleration.z],
-                    [pose.angular_acceleration.x, pose.angular_acceleration.y, pose.angular_acceleration.z],
-                    [pose.mapper_confidence, pose.tracker_confidence],
-                    ])  # raw RealSense2 Pose data
-        return channel
-
-    def run(self):
-        self.pipeline = rs.pipeline()
-        cfg = rs.config()
-        cfg.enable_stream(rs.stream.pose)
-        self.pipeline.start(cfg)
+        self.pose_pipeline = None  # not initialized yet
+        self.depth_pipeline = None
+        self.finished = None
+
+    def pose_callback(self, frame):
+        # TODO: add decimation based on config from 200Hz to desired value
```

Comment:
this is "must have" before merge to master (the robots are at the moment capable to handle 20Hz pose update but not ten times faster)

Reply:
I am well aware of that. As I have stated above:

> Also subsampling is not implemented yet which is meant to replace the timer.tick facility. The idea is to define subsampling to 20 for pose and 3 for depth - we keep every 20th pose and every 3rd depth - resulting in 10fps for both.